specs/agents/logging.md

```diff
@@ -0,0 +1,36 @@
+## Agent logging
+
+### `log_level` configuration
+
+Sets the logging level for the agent.
+
+This option is case-insensitive.
+
+|                |   |
+|----------------|---|
+| Valid options  | `trace`, `debug`, `info`, `warning`, `error`, `critical`, `off` |
+| Default        | `info` |
```

Comment:
Default is `warn` as we don't want logs to be shown by default in production environments.

Reply:
@basepi What was the previous default for the backend agents? Are we ok with changing to waning as the default for them? In the Java agent, we currently log some lightweight info on startup so people know the agent is up and running. Are you suggesting we should get rid of that?

Reply:
I think the default should be `info`- it provides both indication that the agent is attached when everything is fine and some basic info to start analysis of support cases before asking users to run with modified configuration. It should be very non-verbose (we have `debug` and `trace` for those), so shouldn't be a concern in production.

Reply:
The python agent doesn't really have a default, since we don't set up logging ourselves at this point. Should we open this discussion to other agents?

I will note that in general in python I consider INFO production-safe. The idea being that if there's a problem, we go to DEBUG, but INFO provides a base amount of logging that is useful but not too verbose. I can see arguments on either side.

If we went to INFO would that be enough of a problem for RUM as to cause a break in defaults for the RUM agent?

Reply:
This may also be a case where different defaults for RUM and backend agents make sense.
Having said that, the Go and .NET agents currently have a default of `error`.

Reply:
If I were implementing it for the python agent in a vacuum, I'd probably go with `warning` as the default, but would consider `info`. If we default to `info` we may have to do some checking to make sure we're not too verbose.

Reply:
Based on the agents meeting this morning, I think `info` will be the default, but will not be enforced among all the agents. This means that if an agent team thinks `warning` is correct for their agent, they can set that as that agent's default. I don't think the consistency in defaults here is very important.